hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/CrossTenantHandler.java

```diff
@@ -0,0 +1,328 @@
+    })
+    public Response findEvents(
+            @ApiParam(required = false, value = "Filter out events created before this time.",
+                    allowableValues = "Timestamp in millisecond since epoch.")
+            @QueryParam("startTime")
+            final Long startTime,
+            @ApiParam(required = false, value = "Filter out events created after this time.",
+                    allowableValues = "Timestamp in millisecond since epoch.")
+            @QueryParam("endTime")
+            final Long endTime,
+            @ApiParam(required = false, value = "Filter out events for unspecified eventIds.",
+                    allowableValues = "Comma separated list of event IDs.")
+            @QueryParam("eventIds") final String eventIds,
+            @ApiParam(required = false, value = "Filter out events for unspecified triggers.",
+                    allowableValues = "Comma separated list of trigger IDs.")
+            @QueryParam("triggerIds")
+            final String triggerIds,
+            @ApiParam(required = false, value = "Filter out events for unspecified categories. ",
+                    allowableValues = "Comma separated list of category values.")
+            @QueryParam("categories")
+            final String categories,
+            @ApiParam(required = false, value = "Filter out events for unspecified tags.",
+                    allowableValues = "Comma separated list of tags, each tag of format 'name\\|value'. + \n" +
+                            "Specify '*' for value to match all values.")
+            @QueryParam("tags")
+            final String tags,
+            @ApiParam(required = false, value = "Return only thin events, do not include: evalSets.")
+            @QueryParam("thin")
+            final Boolean thin,
+            @Context
+            final UriInfo uri) {
+        Pager pager = RequestUtil.extractPaging(uri);
+        try {
+            Set<String> tenantIds = getTenants(tenantId);
+            EventsCriteria criteria = buildEventsCriteria(startTime, endTime, eventIds, triggerIds, categories, tags, thin);
+            Page<Event> eventPage = alertsService.getEvents(tenantIds, criteria, pager);
+            if (log.isDebugEnabled()) {
+                log.debug("Events: " + eventPage);
+            }
+            if (isEmpty(eventPage)) {
+                return ResponseUtil.ok(eventPage);
+            }
+            return ResponseUtil.paginatedOk(eventPage, uri);
+        } catch (Exception e) {
+            log.debug(e.getMessage(), e);
+            if (e.getCause() != null && e.getCause() instanceof IllegalArgumentException) {
+                return ResponseUtil.badRequest("Bad arguments: " + e.getMessage());
+            }
+            return ResponseUtil.internalError(e);
+        }
+    }
+
+    private AlertsCriteria buildAlertsCriteria(Long startTime, Long endTime, String alertIds, String triggerIds,
```

Comment:
Maybe instead of duplicating these buildAlertsCriteria and buildEventsCriteria we should think about sharing them by making them Static and package private, or even moving this to the AlertsCriteria and EventsCriteria classes as constructors?